File: hanoi.py
import sys

nbMove = 0

def move(ndisks, fromm, via, to):
    if ndisks > 1:
        global nbMove
        nbMove += 1
        move(ndisks - 1, fromm, to, via)
        move(ndisks - 1, via, fromm, to)
 
nbDisks = int(sys.argv[1])
move(nbDisks, 1, 2, 3)
print("Nb of moves: %s" % nbMove)

File: hanoi.ml
let rec hanoi n a b c move =
        if n <> 0 then begin
                Printf.printf "Move disk from pole %d to pole %d\n" a c;
                hanoi (pred n) a c b (succ move);
                hanoi (pred n) b a c (succ move)
        end
 
let () =
  hanoi (int_of_string Sys.argv.(1)) 1 2 3 0
